src/Nethermind/Nethermind.Consensus/Producers/BlockToProduce.cs

```diff
@@ -37,7 +29,9 @@ public new IEnumerable<Transaction> Transactions
             IEnumerable<Withdrawal>? withdrawals = null)
             : base(blockHeader, Array.Empty<Transaction>(), uncles, withdrawals)
         {
-            Transactions = transactions;
+            Transactions = transactions is Transaction[] transactionsArray
+                ? transactionsArray
+                : transactions.ToArray();
```

Comment:
This totally misses the point. We want this to be lazy enumerated as in the end we will only take some part of this enumeration.